autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/CredentialsSync.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.sync
+
+import com.duckduckgo.app.global.formatters.time.DatabaseDateFormatter
+import com.duckduckgo.autofill.api.domain.app.LoginCredentials
+import com.duckduckgo.autofill.store.CredentialsSyncMetadataEntity
+import com.duckduckgo.autofill.sync.provider.LoginCredentialEntry
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.securestorage.api.SecureStorage
+import com.duckduckgo.securestorage.api.WebsiteLoginDetails
+import com.duckduckgo.securestorage.api.WebsiteLoginDetailsWithCredentials
+import com.duckduckgo.sync.api.SyncCrypto
+import dagger.SingleInstanceIn
+import javax.inject.*
+import kotlinx.coroutines.flow.firstOrNull
+import timber.log.Timber
+
+@SingleInstanceIn(AppScope::class)
+class CredentialsSync @Inject constructor(
+    private val secureStorage: SecureStorage,
+    private val credentialsSyncStore: CredentialsSyncStore,
+    private val credentialsSyncMetadata: CredentialsSyncMetadata,
+    private val syncCrypto: SyncCrypto,
+) {
+
+    suspend fun initMetadata() {
+        credentialsSyncStore.serverModifiedSince = "0"
+        credentialsSyncStore.clientModifiedSince = "0"
+
+        secureStorage.websiteLoginDetailsWithCredentials().firstOrNull()?.map { it.details.id }?.filterNotNull()?.let { autofillIds ->
+            credentialsSyncMetadata.initializeDatabase(autofillIds)
+        }
+
+        Timber.i("CredentialsSync: initMetadata ${credentialsSyncMetadata.getAllObservable().firstOrNull()}")
+    }
+
+    suspend fun getUpdatesSince(since: String): List<LoginCredentialEntry> {
```

Comment:
Nit: this could be a good use case for [inline value classes](https://kotlinlang.org/docs/inline-classes.html)

the string parameter could accept any string, but has a very specific format required in order for it to be used. could define something like

`value class Iso8601String(private val s: String)` that is passed into these methods instead of just a string. might help make it clearer what is expected and avoid future errors by forcing the caller to be a little bit more deliberate in the string that is passed

just a suggestion to consider